internal/uvm/scsi.go

```diff
@@ -323,9 +341,40 @@ func (uvm *UtilityVM) addSCSIActual(ctx context.Context, hostPath, uvmPath, atta
 	}
 
 	if err := uvm.modify(ctx, SCSIModification); err != nil {
-		return nil, fmt.Errorf("uvm::AddSCSI: failed to modify utility VM configuration: %s", err)
+		return fmt.Errorf("uvm::AddSCSI: failed to modify utility VM configuration: %s", err)
 	}
-	return sm, nil
+	return nil
+}
+
+// allocateSCSIDevice grants vm access to hostpath and increments the ref count of an existing scsi
+// device or allocates a new one if not already present.
+func (uvm *UtilityVM) allocateSCSIDevice(ctx context.Context, hostPath, uvmPath string, vmAccess VMAccessType) (*SCSIMount, bool, error) {
+	// Ensure the utility VM has access
+	err := uvm.grantAccess(ctx, hostPath, vmAccess)
```

Comment:
Discussed already but maybe find a way to avoid calling this if we've already granted this hostpath access before.

Reply:
I think this should be thought about more and addressed in a later PR. For now, this PR will stand to unblock GPU testing.

Reply:
Sounds good